packages/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx

```diff
@@ -207,27 +210,64 @@ function GridColumnsManagement(props: GridColumnsManagementProps) {
     }
     return false;
   };
+  const handleSearchReset = React.useCallback(() => {
+    setSearchValue('');
+    searchInputRef.current!.focus();
+  }, []);
 
   return (
     <React.Fragment>
       <GridColumnsManagementHeader className={classes.header} ownerState={rootProps}>
         <rootProps.slots.baseTextField
           placeholder={apiRef.current.getLocaleText('columnsManagementSearchTitle')}
+          aria-label={apiRef.current.getLocaleText('columnsManagementSearchTitle')}
           inputRef={searchInputRef}
           value={searchValue}
           onChange={handleSearchValueChange}
           variant="outlined"
           size="small"
+          type="search"
           InputProps={{
             startAdornment: (
               <rootProps.slots.baseInputAdornment position="start">
                 <rootProps.slots.quickFilterIcon />
               </rootProps.slots.baseInputAdornment>
             ),
-            sx: { pl: 1.5 },
+            endAdornment: (
+              <rootProps.slots.baseIconButton
+                aria-label={apiRef.current.getLocaleText('columnsManagementDeleteIconLabel')}
+                size="small"
+                sx={[
+                  searchValue
+                    ? {
+                        visibility: 'visible',
+                      }
+                    : {
+                        visibility: 'hidden',
+                      },
+                ]}
+                tabIndex={-1}
+                onClick={handleSearchReset}
+                {...rootProps.slotProps?.baseIconButton}
+              >
+                <rootProps.slots.quickFilterClearIcon fontSize="small" />
+              </rootProps.slots.baseIconButton>
+            ),
+            sx: {
```

Comment:
I think we try to avoid the `sx` prop where possible in the grid - could we use the [styled](https://mui.com/system/styled/) util for these styles instead?

Reply:
I have shifted those styles to its parent which uses `styled`, it selects the input with type="search" and applies the styles to it.

Reply:
I think it would be better in it's own `styled` for the input itself, it's clearer that the styles belong to that element.

Reply:
I've added it,
also tried using styled only for input in  inputComponent prop inside `InputProps` but it didn't seem to work well.
followed this section of docs https://mui.com/material-ui/react-text-field/#using-the-styled-api

Reply:
Yeah, I think we should have a class on it for overrides.

Reply:
Added
also running pnpm docs:api removes some stuff instead of updating API, is this normal? 
could you check this on your machine?

Reply:
Seems ok for me - I have pushed the changes it generated.

Reply:
thanks, I have resolved the  merge conflicts
would you like to give it a final look before merging

Reply:
<img width="280" alt="Screenshot 2024-11-18 172621" src="https://github.com/user-attachments/assets/948bb965-0444-4f17-be6e-a04b28047a54">